src/main/java/com/epam/lowcost/DAO/implementations/TicketDAOImpl.java

```diff
@@ -146,6 +146,43 @@ public Ticket deleteTicket(long id) {
         return ticket;
     }
 
+    public boolean deleteTicketsByFlightId(long id) {
+        int required, updated;
+        try {
+            JdbcTemplate jdbcTemplate = new JdbcTemplate(dataSource);
+            String query = "Select * FROM Tickets JOIN  USERS ON TICKETS.userId=USERS.id " +
+                    "JOIN  FLIGHTS ON TICKETS.flightId=FLIGHTS.id " +
+                    "JOIN  PLANES ON FLIGHTS.planeId = PLANES.id " +
+                    "WHERE TICKETS.isDeleted=false AND TICKETS.FLIGHTID = ?";
+            List<Ticket> tickets = jdbcTemplate.query(query, ticketRowMapper, id);
+            required = tickets.size();
```

Comment:
wtf ? select * ? why ?

Reply:
compare all rows that SHOULD be updated with those that ARE updated